beacon-chain/db/filesystem/data_column.go

```diff
@@ -0,0 +1,966 @@
+}
+
+// WarmCache warms the cache of the data column filesystem.
+// It holds the database (read) lock for all the time it is running.
+func (dcs *DataColumnStorage) WarmCache() {
+	start := time.Now()
+	log.Info("Data column filesystem cache warm-up started")
+
+	dcs.pruneMu.Lock()
+	defer dcs.pruneMu.Unlock()
+
+	highestStoredEpoch := primitives.Epoch(0)
+
+	// Walk the data column filesystem to warm up the cache.
+	if err := afero.Walk(dcs.fs, ".", func(path string, info os.FileInfo, fileErr error) (err error) {
+		if fileErr != nil {
+			return fileErr
+		}
+
+		// If not a leaf, skip.
+		if info.IsDir() {
+			return nil
+		}
+
+		// Extract metadata from the file path.
+		fileMetadata, err := extractFileMetadata(path)
+		if err != nil {
+			log.WithError(err).Error("Error encountered while extracting file metadata")
+		}
+
+		// Open the data column filesystem file.
+		file, err := dcs.fs.Open(path)
+		if err != nil {
+			log.WithError(err).Error("Error encountered while opening data column filesystem file")
+			return nil
+		}
+
+		// Close the file.
+		defer func() {
+			// Overwrite the existing error only if it is nil, since the close error is less important.
+			closeErr := file.Close()
+			if closeErr != nil && err == nil {
+				err = closeErr
+			}
+		}()
+
+		// Read the metadata of the file.
+		metadata, err := dcs.metadata(file)
+		if err != nil {
+			log.WithError(err).Error("Error encountered while reading metadata from data column filesystem file")
+			return nil
+		}
+
+		// Check the indices.
+		indices := make([]uint64, 0, len(metadata.indices))
+		for index, i := range metadata.indices {
+			// Skip if the data column is not saved.
+			if i < limit {
+				continue
+			}
+
+			indices = append(indices, uint64(index))
+		}
+
+		if len(indices) == 0 {
+			return nil
+		}
+
+		// Build the ident.
+		dataColumnsIdent := DataColumnsIdent{Root: fileMetadata.blockRoot, Epoch: fileMetadata.epoch, Indices: indices}
+
+		// Update the highest stored epoch.
+		highestStoredEpoch = max(highestStoredEpoch, fileMetadata.epoch)
+
+		// Set the ident in the cache.
+		if err := dcs.cache.set(dataColumnsIdent); err != nil {
+			log.WithError(err).Error("Error encountered while ensuring data column filesystem cache")
+		}
+
+		return nil
+	}); err != nil {
+		log.WithError(err).Error("Error encountered while walking data column filesystem.")
+	}
+
+	// Prune the cache and the filesystem.
+	dcs.prune()
+
+	log.WithField("elapsed", time.Since(start)).Info("Data column filesystem cache warm-up complete")
+}
+
+// Summary returns the DataColumnStorageSummary.
+func (dcs *DataColumnStorage) Summary(root [fieldparams.RootLength]byte) DataColumnStorageSummary {
+	return dcs.cache.Summary(root)
+}
+
+// Store stores data column sidecars into the database and asynchronously performs pruning.
+// The returned chanel is closed when the pruning is complete.
+func (dcs *DataColumnStorage) Store(dataColumnSidecars []blocks.VerifiedRODataColumn) error {
```

Comment:
nitpick: `Store` is fine but we do often use `store` as a noun for something that abstracts storage. `Save` would avoid that ambiguity and be consistent with the blob store.

Reply:
Fixed in https://github.com/prysmaticlabs/prysm/pull/15061/commits/799ed429ca63a61e9aa53b8c9de257efca3d9eb2.